Modules/xSharePoint/DSCResources/MSFT_xSPFeature/MSFT_xSPFeature.psm1

```diff
@@ -80,12 +114,20 @@ function Test-TargetResource
         [parameter(Mandatory = $true)]  [System.String] $Url,
         [parameter(Mandatory = $true)]  [ValidateSet("Farm","WebApplication","Site","Web")] [System.String] $FeatureScope,
         [parameter(Mandatory = $false)] [System.Management.Automation.PSCredential] $InstallAccount,
-        [parameter(Mandatory = $true)]  [ValidateSet("Present","Absent")] [System.String] $Ensure
+        [parameter(Mandatory = $true)]  [ValidateSet("Present","Absent")] [System.String] $Ensure,
+        [parameter(Mandatory = $false)] [System.String] $Version
     )
 
     $CurrentValues = Get-TargetResource @PSBoundParameters
     Write-Verbose -Message "Testing for feature $Name at $FeatureScope scope"
-    return Test-xSharePointSpecificParameters -CurrentValues $CurrentValues -DesiredValues $PSBoundParameters -ValuesToCheck @("Ensure")
+
+    $valuesToCheck = @("Ensure")
+
+    if ($Version) {
+        $valuesToCheck += @("Version")
+    }
+
+    return Test-xSharePointSpecificParameters -CurrentValues $CurrentValues -DesiredValues $PSBoundParameters -ValuesToCheck $valuesToCheck
```

Comment:
Can you pass version in here anyway to keep it clean - if version is not specified in PSBoundParameters then it will automatically be ignored by the Test-xSharePointSpecificParameters method, so just take the line as it was and add the "Version" bit to the existing array next to Ensure